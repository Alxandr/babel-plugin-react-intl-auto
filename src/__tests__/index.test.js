// @flow
import path from 'path'
import fs from 'fs'
import { transformFileSync } from 'babel-core'

import plugin from '../'

const testPlugin = (filePath, opts = {}) => {
  const { code } = transformFileSync(filePath, {
    plugins: [[plugin, opts]],
  })
  return code
}

describe('snapshot', () => {
  const rootPath = path.resolve(__dirname, '../__fixtures__')
  const files = fs.readdirSync(rootPath)

  for (const file of files) {
    test(`snapshot -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file))
      expect(result).toMatchSnapshot()
    })

    test(`removePrefix -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        removePrefix: 'src/',
      })
      expect(result).toMatchSnapshot()
    })

    test(`removePrefix:no-slash -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        removePrefix: 'src',
      })
      expect(result).toMatchSnapshot()
    })
  }
})
