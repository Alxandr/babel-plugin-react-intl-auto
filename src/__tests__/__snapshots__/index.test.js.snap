// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot removePrefix -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot removePrefix -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot removePrefix:no-slash -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot snapshot -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot snapshot -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot snapshot -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot snapshot -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;
